src/hugr.rs

```diff
@@ -193,7 +193,7 @@ impl Hugr {
         rw.apply(self)
     }
 
-    /// Infer extension requirements
+    /// Infer extension requirements and add new information to `op_types` field
```

Comment:
Add a doclink to `infer::infer_extensions` here?
And, should the return type here be `ExtensionSolution`? (A typedef to the same type as here)

Reply:
I think it should be, yes. I was trying to distinguish the genuine solution (as `ExtensionSolution`) from the closure, but they really should be the same type regardless

Reply:
I've now addressed both of these in `8e55229` and `1802ccd`, respectively